aws/nat_gateway.go

```diff
@@ -0,0 +1,107 @@
+package aws
+
+import (
+	"time"
+
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+)
+
+func waitUntilNatGatewayDeleted(svc *ec2.EC2, input *ec2.DescribeNatGatewaysInput) error {
+	for i := 0; i < 30; i++ {
+		result, err := svc.DescribeNatGateways(input)
+		if err != nil {
+			if awsErr, isAwsErr := err.(awserr.Error); isAwsErr && awsErr.Code() == "NatGatewayNotFound" {
+				return nil
+			}
+
+			return err
+		}
+
+		if *result.NatGateways[0].State == "deleted" {
+			return nil
+		}
+
+		time.Sleep(5 * time.Second)
+		logging.Logger.Debug("Waiting for NAT Gateway to be deleted")
```

Comment:
To make the logs easier to follow, move log statement before sleep and say something like, "NAT Gateway still not deleted. Will sleep for 5 seconds and check again."

Reply:
ok